src/Raven.Server/Documents/Handlers/Processors/Subscriptions/AbstractSubscriptionsHandlerProcessorForCreateSubscription.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client.Documents.Subscriptions;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers.Processors.Subscriptions
+{
+    internal abstract class AbstractSubscriptionsHandlerProcessorForCreateSubscription<TRequestHandler, TOperationContext> : AbstractSubscriptionsHandlerProcessorForCreateOrUpdateSubscription<TRequestHandler, TOperationContext>
+        where TOperationContext : JsonOperationContext
+        where TRequestHandler : AbstractDatabaseRequestHandler<TOperationContext>
+    {
+        protected AbstractSubscriptionsHandlerProcessorForCreateSubscription([NotNull] TRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected override bool GetDisabled(SubscriptionCreationOptions options)
+        {
+            return options.Disabled;
+        }
+
+        protected override long? GetSubscriptionId(SubscriptionUpdateOptions options)
+        {
```

Comment:
do we need to consider the `SubscriptionUpdateOptions options` here?
should we call the `base` ?

Reply:
no

Reply:
https://github.com/ravendb/ravendb/blob/ca2bbebffa249b553c20ec9d054c9014f83642b0/src/Raven.Server/Documents/Handlers/Processors/Subscriptions/AbstractSubscriptionsHandlerProcessorForPutSubscription.cs#L32